web/modules/custom/dpl_related_content/dpl_related_content.module

```diff
@@ -0,0 +1,292 @@
+<?php
+
+use Drupal\Core\Database\Database;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\Entity\FieldableEntityInterface;
+use Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface;
+use Drupal\node\Entity\Node;
+use Drupal\node\NodeInterface;
+use Drupal\recurring_events\Entity\EventInstance;
+
+/**
+ * Get terms from field as an array of IDs.
+ *
+ * @return array<int>
+ *   Term IDs.
+ */
+function _dpl_related_content_get_term_ids(FieldableEntityInterface $entity, string $field_name): array {
+  if (!$entity->hasField($field_name)) {
+    return [];
+  }
+
+  $terms = $entity->get($field_name)->getValue();
+  return array_column($terms, 'target_id');
+
+}
+
+/**
+ * Get matching eventinstance IDs.
+ *
+ * Allows for passing along various term IDs, that we look for in an OR group.
+ * Notice - we only look for the term values on series level, ignoring
+ * inheritance overriding in favor of a simpler codebase.
+ *
+ * @param string|null $current_uuid
+ *   A possible entity UUID, that will not get included in results.
```

Comment:
Consinder changing this here and elsewhere to ids instead of.

I wonder why this is a UUID when all the other values we pass around here are ids. This also goes elsewhere in this PR.

Reply:
I'll add a comment in the code why - here, for your convienience:

```
   *   We use UUID instead of IDs, as UUID will be unique across entity types,
   *   and means we don't need to worry about sending a node UUID along to a
   *   event query.